jenkins-build-pack-website/Jenkinsfile

```diff
@@ -47,212 +46,210 @@ node()  {
+				validateDeploymentConfigurations()
 			} catch (ex) {
-				send_status_email('FAILED', config, '')
+				send_status_email('FAILED', '')
 				events.sendFailureEvent('VALIDATE_PRE_BUILD_CONF', ex.getMessage(), context_map)
 				error ex.getMessage()
 			}
 			events.sendCompletedEvent('VALIDATE_PRE_BUILD_CONF', 'pre-build validation completed', context_map)
 		}
 
-		def s3StgBucket  = configLoader.JAZZ.S3.WEBSITE_STG_BUCKET;
-		def s3ProdBucket = configLoader.JAZZ.S3.WEBSITE_PROD_BUCKET;
-		def s3DevBucket  = configLoader.JAZZ.S3.WEBSITE_DEV_BUCKET;
+		if (configLoader.JAZZ.BUCKET_PER_SERVICE == "true") {
+			if (service_config['s3_bucket_name']) {
+				s3Bucket = service_config['s3_bucket_name']
+			} else {
+				s3Bucket = "${configLoader.INSTANCE_PREFIX}-${utilModule.generateBucketNameForService(service_config['domain'], service_config['service'])}"
+				def serviceMetadataJson = ["s3_bucket_name" : s3Bucket]
+                for (item in service_config.catalog_metadata) {
+					serviceMetadataJson[item.key] = item.value
+				}
+				context_map['metadata'] = serviceMetadataJson
+			}
+			asset_info = "${s3Bucket}/${environment}"
+			asset_name = "${s3Bucket}/${environment}"
+
+		} else {
+			asset_info = "${s3Bucket}/${service}/${environment}"
+			asset_name = "${s3Bucket}/${service}"
+		}
 
-		stage("Deployment to ${environment} Env"){
-				events.sendStartedEvent("DEPLOY_TO_AWS", 'deployment starts', context_map)
-				events.sendStartedEvent('UPDATE_ENVIRONMENT', 'Environment  status update event  for ${environment} deployment', environmentDeploymentMetadata.generateEnvironmentMap("deployment_started", environment), environment)
-				withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: configLoader.AWS_CREDENTIAL_ID, secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
-					echo "AWS Configure ......."
+		stage("Deployment to ${current_environment} Env"){
+			events.sendStartedEvent("DEPLOY_TO_AWS", 'deployment starts', context_map)
+			events.sendStartedEvent('UPDATE_ENVIRONMENT', 'Environment  status update event  for ${environment} deployment', environmentDeploymentMetadata.generateEnvironmentMap("deployment_started", environment), environment)
+			withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: configLoader.AWS_CREDENTIAL_ID, secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+				echo "AWS Configure ......."
+
+				sh "aws configure set profile.cloud-api.region ${configLoader.AWS.REGION}"
+				sh "aws configure set profile.cloud-api.aws_access_key_id $AWS_ACCESS_KEY_ID"
+				sh "aws configure set profile.cloud-api.aws_secret_access_key $AWS_SECRET_ACCESS_KEY"
+				sh "aws configure set preview.cloudfront true"
+
+				def assetInfo = "S3 Folder - /${asset_info}"
+
+				echo "Checking if the website already exists.."
+				
+				if (configLoader.JAZZ.BUCKET_PER_SERVICE == "true") {
```

Comment:
There is a lot of overlapping code between If and else? Can you please take a look at both and refactor it in such a way that we can drive the code through a common function with the right arguments?

Reply:
done